mongodb_store/scripts/replicator_node.py

```diff
@@ -139,9 +148,21 @@ def do_dump(self, collection, master, less_time_time=None, db='message_store'):
 
         # print args
 
-        subprocess.call(args)
-
-
+        self.dump_process = subprocess.Popen(args)
+        self.dump_process.wait()
+
+
+    def do_cancel(self):
+        if self.restore_process and self.restore_process.poll() is None:
```

Comment:
@hawesie Sorry, I did not what you mean. Yes, to check if these are not None. I'll change these parts.